corehq/form_processor/abstract_models.py

```diff
@@ -192,6 +192,31 @@ def get_properties_in_api_format(self):
             # all custom properties go here
         }.items())
 
+    @property
+    @memoized
+    def parent(self):
+        from casexml.apps.case.models import INDEX_ID_PARENT
+        from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
+        """
+        Returns the parent case if one exists, else None.
+        NOTE: This property should only return the first parent in the list
+        of indices. If for some reason your use case creates more than one,
+        please write/use a different property.
+        """
+        for index in self.indices:
+            if index.identifier == INDEX_ID_PARENT:
+                return CaseAccessors(self.domain).get_case(index.referenced_id)
+        return None
+
+    @memoized
+    def get_attachment_map(self):
+        return dict([
+            (name, {
+                'url': self.get_attachment_server_url(att.attachment_key),
```

Comment:
Since `attachment_key` is just an alias for `identifier` and only used in a few 2 places I'd vote to do away with the alias.